graylog2-server/src/main/java/org/graylog/events/processor/DBEventDefinitionService.java

```diff
@@ -65,13 +65,23 @@ public EventDefinitionDto saveWithOwnership(EventDefinitionDto eventDefinitionDt
         return dto;
     }
 
-    public int deleteAndUnregister(String id) {
+    public int deleteAndUnregister(String id, boolean overrideMutable) {
```

Comment:
@roberto-graylog @ryan-carroll-graylog Do you have any opinion if I should break out a separate method `deleteAndUnregisterMutable` and avoid the boolean parameter here? I was not certain if one is better than the other.

Reply:
I'm in favor of breaking it out to make it more explicit and less likely to accidently use the wrong way. Plus that's the convention of the superclass.

Reply:
Thanks for the advice @ryan-carroll-graylog. Will do.

Reply:
Thanks again @ryan-carroll-graylog. Done in https://github.com/Graylog2/graylog2-server/pull/13095/commits/468e761780205ea17edbaaf5d615c9aeffeb1e29. Let me know if you have any feedback on the implementation.

I like the API better this way (vs the previous boolean approach). It's more expressive and preserves the existing `delete` API method signatures (which we typically want to do in core.).